Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/ErrorHandlingTransmissionPolicyTest.cs

```diff
@@ -97,6 +100,249 @@ public void ResumesTransmissionSenderAfterPauseDuration()
             }
 
             [TestMethod]
+            public void ForPartialSuccessIfItemIsRejectedOnlyThisItemIsUploadedBack()
+            {
+                IList<Transmission> enqueuedTransmissions = new List<Transmission>();
+                var transmitter = new StubTransmitter
+                {
+                    OnEnqueue = t => { enqueuedTransmissions.Add(t); }
+                };
+
+                var policy = new ErrorHandlingTransmissionPolicy();
+                policy.Initialize(transmitter);
+
+                var items = new List<ITelemetry> { new EventTelemetry(), new EventTelemetry() };
+                Transmission transmission = new Transmission(new Uri("http://uri"), items, "type", "encoding");
+
+                string breezeResponse =
+                   "{" +
+                    "\"itemsReceived\": 2," +
+                    "\"itemsAccepted\": 1," +
+                    "\"errors\": [" +
+                                    "{" +
+                                      "\"index\": 0," +
+                                      "\"statusCode\": 402," +
+                                      "\"message\": \"Explanation\"" +
+                                    "}" +
+                   "]}";
+
+                transmitter.OnTransmissionSent(new TransmissionProcessedEventArgs(transmission, null, breezeResponse));
+
+                Assert.Equal(1, enqueuedTransmissions.Count);
+            }
+
+            [TestMethod]
+            public void ForPartialSuccessIfItemsAreRejectedTheyAreUploadedBackGroupedByStatusCode()
+            {
+                IList<Transmission> enqueuedTransmissions = new List<Transmission>();
+                var transmitter = new StubTransmitter
+                {
+                    OnEnqueue = t => { enqueuedTransmissions.Add(t); }
+                };
+
+                var policy = new ErrorHandlingTransmissionPolicy();
+                policy.Initialize(transmitter);
+
+                var items = new List<ITelemetry> { new EventTelemetry(), new EventTelemetry() };
+                Transmission transmission = new Transmission(new Uri("http://uri"), items, "type", "encoding");
+
+                string breezeResponse =
```

Comment:
I think it would be more difficult to understand what test is doing if I add helper methods, parameters, start using string.format and add more elements in "items" depending on the parameter...